worlds/oot_soh/location_access/dungeons/fire_temple.py

```diff
@@ -2,8 +2,518 @@
+        (Regions.FIRE_TEMPLE_FIRE_PILLAR_ROOM, lambda bundle: small_keys(Items.FIRE_TEMPLE_SMALL_KEY, 3, bundle))
+    ])
+    
+    ## Fire Temple Big Lava Room North Goron
+    # Locations
+    add_locations(Regions.FIRE_TEMPLE_BIG_LAVA_ROOM_NORTH_GORON, world, [
+        (Locations.FIRE_TEMPLE_BIG_LAVA_ROOM_LOWER_OPEN_DOOR_CHEST, lambda bundle: True)
+    ])
+    # Connections
+    connect_regions(Regions.FIRE_TEMPLE_BIG_LAVA_ROOM_NORTH_GORON, world, [
+        (Regions.FIRE_TEMPLE_BIG_LAVA_ROOM, lambda bundle: True)
+    ])
+    
+    ## Fire Temple Big Lava Room North Tiles
+    # Locations
+    add_locations(Regions.FIRE_TEMPLE_BIG_LAVA_ROOM_NORTH_TILES, world, [
+        (Locations.FIRE_TEMPLE_GS_SONG_OF_TIME_ROOM, lambda bundle:
+            ((is_adult(bundle) and
+              can_attack(bundle)) or
+             hookshot_or_boomerang(bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.FIRE_TEMPLE_BIG_LAVA_ROOM_NORTH_TILES, world, [
+        (Regions.FIRE_TEMPLE_BIG_LAVA_ROOM, lambda bundle: True)
+    ])
+    
+    ## Fire Temple Big Lava Room South Goron
+    # Locations
+    add_locations(Regions.FIRE_TEMPLE_BIG_LAVA_ROOM_SOUTH_GORON, world, [
+        (Locations.FIRE_TEMPLE_BIG_LAVA_ROOM_BLOCKED_DOOR_CHEST, lambda bundle: True)
+    ])
+    # Connections
+    connect_regions(Regions.FIRE_TEMPLE_BIG_LAVA_ROOM_SOUTH_GORON, world, [
+        (Regions.FIRE_TEMPLE_BIG_LAVA_ROOM, lambda bundle: True)
+    ])
+    
+    ## Fire Temple Fire Pillar Room
+    # Locations
+    add_locations(Regions.FIRE_TEMPLE_FIRE_PILLAR_ROOM, world, [
+        (Locations.FIRE_TEMPLE_FIRE_PILLAR_ROOM_LEFT_HEART, lambda bundle: fire_timer(bundle) >= 56),
+        (Locations.FIRE_TEMPLE_FIRE_PILLAR_ROOM_RIGHT_HEART, lambda bundle: fire_timer(bundle) >= 56),
+        (Locations.FIRE_TEMPLE_FIRE_PILLAR_ROOM_BACK_HEART, lambda bundle: fire_timer(bundle) >= 56),
+    ])
+    # Connections
+    connect_regions(Regions.FIRE_TEMPLE_FIRE_PILLAR_ROOM, world, [
+        (Regions.FIRE_TEMPLE_BIG_LAVA_ROOM, lambda bundle: small_keys(Items.FIRE_TEMPLE_SMALL_KEY, 3, bundle)),
+        (Regions.FIRE_TEMPLE_SHORTCUT_ROOM, lambda bundle:
+            (fire_timer(bundle) >= 56 and
+             small_keys(Items.FIRE_TEMPLE_SMALL_KEY, 4, bundle)))
+    ])
+    
+    ## Fire Temple Shortcut Room
+    # Connections
+    connect_regions(Regions.FIRE_TEMPLE_SHORTCUT_ROOM, world, [
+        (Regions.FIRE_TEMPLE_FIRE_PILLAR_ROOM, lambda bundle: small_keys(Items.FIRE_TEMPLE_SMALL_KEY, 4, bundle)),
+        (Regions.FIRE_TEMPLE_SHORTCUT_CLIMB, lambda bundle: True),
+        (Regions.FIRE_TEMPLE_BOULDER_MAZE_LOWER, lambda bundle:
+            (is_adult(bundle) and
+             (has_item(Items.GORONS_BRACELET, bundle) or
+              can_do_trick(Tricks.FIRE_STRENGTH, bundle)) and
+             (has_explosives(bundle) or
+              can_use(Items.FAIRY_BOW) or
+              can_use(Items.HOOKSHOT, bundle) or
+              can_use(Items.FAIRY_SLINGSHOT))))
+    ])
```

Comment:
This is also missing can_ground_jump

Reply:
could also be simplified 
```suggestion
            (is_adult(bundle) and
             (has_item(Items.GORONS_BRACELET, bundle) or
              can_do_trick(Tricks.FIRE_STRENGTH, bundle)) and
             (has_explosives(bundle) or
              can_use_any([Items.FAIRY_BOW,Items.HOOKSHOT,Items.FAIRY_SLINGSHOT], bundle))))
    ])
```

Reply:
Also some of your can_use calls are missing bundle